src/Microsoft.Data/DataFrame.BinaryOperations.cs

```diff
@@ -14,419 +14,503 @@ public partial class DataFrame
     {
         #region Binary Operations
 
-        public DataFrame Add<T>(IReadOnlyList<T> values)
+        public DataFrame Add<T>(IReadOnlyList<T> values, bool inPlace = false)
             where T : unmanaged
         {
             if (values.Count != ColumnCount)
             {
-                throw new ArgumentException($"values.Count {values.Count} must match the number of columns in the table", nameof(values));
+                throw new ArgumentException(Strings.MismatchedColumnLengths, nameof(values));
```

Comment:
(minor) it may be nice to log both `values.Count` and `ColumnCount` here in the exception message. That way when this occurs people can see the values without having to debug it.

Reply:
I'll do this slowly in the other PRs. I think I need to refactor and possibly split this message.